http/src/request/guild/create_guild.rs

```diff
@@ -67,6 +71,104 @@ struct CreateGuildFields {
     verification_level: Option<VerificationLevel>,
 }
 
+/// Fillable struct to send new roles to a guild create.
+#[derive(Clone, Debug, Default)]
+pub struct RoleFragment {
+    pub color: Option<u32>,
+    pub hoist: bool,
+    pub id: Option<RoleId>,
+    pub mentionable: bool,
+    pub name: Option<String>,
+    pub permissions: Option<Permissions>,
+    pub position: Option<i64>,
+}
+
+#[derive(Clone, Debug)]
+pub enum GuildChannelFragment {
+    Category(CategoryChannelFragment),
+    Text(TextChannelFragment),
+    Voice(VoiceChannelFragment),
+}
+
+/// Fillable struct to send new category channels to a guild create.
+#[derive(Clone, Debug)]
+pub struct CategoryChannelFragment {
+    pub id: ChannelId,
+    pub name: String,
+    pub permission_overwrites: Option<Vec<PermissionOverwrite>>,
+}
+
+impl Into<CategoryChannel> for CategoryChannelFragment {
+    fn into(self) -> CategoryChannel {
+        CategoryChannel {
+            guild_id: Some(GuildId(1)),
+            id: self.id,
+            kind: ChannelType::GuildCategory,
+            name: self.name,
+            nsfw: false,
+            parent_id: None,
+            permission_overwrites: self.permission_overwrites.unwrap_or_else(Vec::new),
+            position: 0,
+        }
+    }
+}
+
+/// Fillable struct to send new text channels to a guild create.
+#[derive(Clone, Debug)]
+pub struct TextChannelFragment {
+    pub name: String,
+    pub nsfw: bool,
+    pub parent_id: Option<ChannelId>,
+    pub permission_overwrites: Option<Vec<PermissionOverwrite>>,
+    pub rate_limit_per_user: Option<u64>,
+    pub topic: Option<String>,
+}
+
+impl Into<TextChannel> for TextChannelFragment {
+    fn into(self) -> TextChannel {
+        TextChannel {
+            guild_id: Some(GuildId(1)),
+            id: ChannelId(501),
+            kind: ChannelType::GuildText,
+            last_message_id: None,
+            last_pin_timestamp: None,
+            name: self.name,
+            nsfw: self.nsfw,
+            permission_overwrites: self.permission_overwrites.unwrap_or_else(Vec::new),
+            parent_id: self.parent_id,
+            position: 0,
+            rate_limit_per_user: self.rate_limit_per_user,
+            topic: self.topic,
+        }
+    }
+}
+
+/// Fillable struct to send new voice channels to a guild create.
+#[derive(Clone, Debug)]
+pub struct VoiceChannelFragment {
+    pub bitrate: Option<u64>,
+    pub name: String,
+    pub parent_id: Option<ChannelId>,
+    pub permission_overwrites: Option<Vec<PermissionOverwrite>>,
+    pub user_limit: Option<u64>,
+}
+
+impl Into<VoiceChannel> for VoiceChannelFragment {
+    fn into(self) -> VoiceChannel {
+        VoiceChannel {
+            bitrate: self.bitrate.unwrap_or(64000),
+            guild_id: Some(GuildId(1)),
+            id: ChannelId(502),
```

Comment:
I guess the same as the text channel, does a voice channel's ID being one more than the text channel's 501 have a meaning?